### PR TITLE
feat dashboards

### PR DESCRIPTION
**Pull request by tracy-french**
wip

# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: apps/client/src/assets/messages/index.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { messages } from './messages-en';
```
Comment (boweihan) [inline apps/client/src/assets/messages/index.ts:1]: 👍

File: apps/client/src/index.tsx [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
+import { router } from './router';
+
+const awsResources = (global as any).awsResources;
+
+Amplify.configure({
+  ...awsResources,
+});
 
-import { router } from './router/router';
+applyMode(Mode.Dark);
```
Comment (boweihan) [inline apps/client/src/index.tsx:16]: will switching this to Mode.Light just "work" or will we need to do styling tweaks?

File: apps/client/src/routes/dashboards/components/delete-modal/delete-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            <Button variant="primary" onClick={onDelete}>
+              {messages.delete}
+            </Button>
+          </SpaceBetween>
+        </Box>
+      }
+    >
+      <SpaceBetween size="m">
+        <Box variant="span">
+          Permanently delete dashboard{' '}
```
Comment (boweihan) [inline apps/client/src/routes/dashboards/components/delete-modal/delete-modal.tsx:35]: `messages`? :)

File: apps/client/src/routes/root.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+import { INDEX_ROUTE } from '.';
+import { DASHBOARDS_ROUTE } from './dashboards/dashboards';
+import { TopNavigation } from '@cloudscape-design/components';
+//import { Authenticator } from '@aws-amplify/ui-react';
+
+export const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      staleTime: 1000 * 10,
```
Comment (boweihan) [inline apps/client/src/routes/root.tsx:17]: neat! looks like tanstack query gives us quite a bit of power

File: apps/client/src/services/generated/core/CancelablePromise.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
+/* istanbul ignore file */
```
Comment (boweihan) [inline apps/client/src/services/generated/core/CancelablePromise.ts:1]: this is some spicy codegen - what's the process to generate? Is this something we need to include as part of `npm build`?

File: apps/client/package.json [hunk 6]
```diff
@@ -8,21 +8,29 @@
...
     "@cloudscape-design/global-styles": "^1.0.6",
     "@iot-app-kit/dashboard": "^3.0.0-alpha.3",
     "@iot-app-kit/source-iotsitewise": "^2.6.5",
-    "core-types": "*",
-    "lib": "*",
+    "@tanstack/react-query": "^4.24.10",
+    "@tanstack/react-query-devtools": "^4.24.14",
+    "aws-amplify": "^5.0.17",
+    "openapi-typescript": "^6.1.1",
+    "openapi-typescript-codegen": "^0.23.0",
```
Comment (boweihan) [inline apps/client/package.json:28]: Should these be dev dependencies?

File: apps/client/src/index.tsx (hunk 2 above)
Comment (tracy-french) [reply apps/client/src/index.tsx:23]: It will just work. :)

**Pull request closed without merging**